rskj-core/src/test/java/co/rsk/net/sync/BlockConnectorHelperTest.java

```diff
@@ -0,0 +1,125 @@
+package co.rsk.net.sync;
+
+import co.rsk.core.BlockDifficulty;
+import org.ethereum.core.Block;
+import org.ethereum.db.BlockStore;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.Mockito.*;
+
+class BlockConnectorHelperTest {
```

Comment:
I see that this is covering the whole class, but instead of use so much mocked classes, wdyt in use real blocks and difficulties? The problem in use so much mocked things is that if you change the behavior insise your class and isn't connecting properly, your test would still pass because you are basically setting up everything with mocked parameters. For example, I have changed the code to save always the difficulty from the first block instead get the current one. The test passed the same way, it didn't fail. If you were using real blocks and difficulties, you could compare the passed to save with the saved ones on the blockStore. For more context, this [google article](https://testing.googleblog.com/2013/05/testing-on-toilet-dont-overuse-mocks.html) talks about the issues with so much mocking. I know it's harder to test and write tests without so much mocking, some times takes more time than the code itself, but...it brings more benefit.

We have some auxiliar classes to generate blocks for us, difficulties and you can even use a real blockStore without have to mock it. See the class `BlockGenerator` for possibilities of generate blocks.
If you want also some inspiration in how to use it even with a blockStore and everything, see the test class `SnapBlocksResponseMessageTest` and the `SnapshotProcessorTest` class. I have created some examples there.

Reply:
I'm not totally agree with this comment. The aim of this class is to test the behaviour of the BlockConnectorHelper, so for me it is ok with mocking all the external behaviours. 
Generating blocks os using a blockStore will make this tests heavier using more resources and more time to execute them when it is not really needed as the tests with "real data" will be done within the integration/end-to-end tests.

In other hand for this tests I totally miss to check the behaviour of the difficulties, assuming that the tested class will use difficulty of the same object. I will update this test to make it more accurate.

Reply:
Adding difficulties to the test fixed a bug 😅

Reply:
Maybe we can go with a new integration test and let these tests in place. WDYT guys?

Reply:
Yes, this topic was already discussed, I am resolving the conversation and let the final call with @asoto-iov. 🙂